### PR TITLE
ESP32: Support applying patch to local IDF

### PR DESCRIPTION
**Pull request by tangzz98**
## 拉取/合并请求描述：(PR description)

[
支持在本地ESP-IDF打patch，不用下载软件包
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [ ] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

File: bsp/ESP32_C3/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,5 @@
-# The following lines of boilerplate have to be in your project's
-# CMakeLists in this exact order for cmake to work correctly
 cmake_minimum_required(VERSION 3.16)
 set(COMPONENTS esptool_py main)
 include($ENV{IDF_PATH}/tools/cmake/project.cmake)
+set(freertos_root /Users/tangyi/rt-thread/bsp/ESP32_C3/packages/FreeRTOS_Wrapper-latest/FreeRTOS)
```
Comment (mysterywolf) [inline bsp/ESP32_C3/CMakeLists.txt:4]: 这个路径有问题

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/ESP32_C3/drivers/Kconfig [hunk 2]
```diff
@@ -7,11 +7,15 @@ config SOC_ESP32_C3
     select SOC_ESPRESSIF
     select RT_USING_COMPONENTS_INIT
     select RT_USING_USER_MAIN
-    select PKG_USING_ESP_IDF
+    select PKG_USING_ESP_IDF if BSP_USING_LOCAL_ESP_IDF = n
```
Comment (mysterywolf) [inline bsp/ESP32_C3/drivers/Kconfig:10]: 有这种写法嘛？

File: bsp/ESP32_C3/CMakeLists.txt (hunk 1 above)
Comment (tangzz98) [reply bsp/ESP32_C3/CMakeLists.txt:4]: 这是scons --target=esp-idf自动生成的。本地的IDF可以放在任何地方，这边必须写绝对路径

File: bsp/ESP32_C3/drivers/Kconfig (hunk 2 above)
Comment (tangzz98) [reply bsp/ESP32_C3/drivers/Kconfig:10]: 我测试没问题

File: bsp/ESP32_C3/CMakeLists.txt (hunk 1 above)
Comment (mysterywolf) [reply bsp/ESP32_C3/CMakeLists.txt:4]: 哦哦好的

Comment (supperthomas) [inline bsp/ESP32_C3/CMakeLists.txt:4]: 这里是绝对路径？是自动生成的么？

**Review by supperthomas — APPROVED** [review]

File: bsp/ESP32_C3/CMakeLists.txt (hunk 1 above)
Comment (tangzz98) [reply bsp/ESP32_C3/CMakeLists.txt:4]: 是的

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**